flexget/plugins/input/anidb_feed.py

```diff
@@ -0,0 +1,453 @@
+                    if len(dt_nodes) != len(dd_nodes):
+                        log.trace('unexpected nodecount')
+                        continue
+                    for idx, dt_node in enumerate(dt_nodes):
+                        if len(dt_node.childNodes) >= 1 and len(dd_nodes[idx].childNodes) >= 1:
+                            if dt_node.childNodes[0].nodeValue is not None and dd_nodes[idx].childNodes[0].nodeValue is not None:
+                                dt_node.tagName = NAMESPACE_NAME + ':%s' % dt_node.childNodes[0].nodeValue.replace(' ', '_')
+                                dt_node.setAttribute('name', dt_node.childNodes[0].nodeValue)
+                                dt_node.setAttribute('value', dd_nodes[idx].childNodes[0].nodeValue)
+                                dt_node.removeChild(dt_node.childNodes[0])
+                                node_ns.removeChild(dd_nodes[idx])
+                                entry.appendChild(dt_node)
+        return dom.toxml()
+
+    def _set_ns_attribute(self, name, xml_entry, entry, in_ns_name, out_prefix_name=None, in_type=str):
+        tagname = in_ns_name + '_' + name  # feedparser represents those as nsname_tagname
+        value = None
+        if tagname in entry and entry.get(tagname) is not None:
+            value = entry.get(tagname)  # use existing, but do type check
+        elif tagname in xml_entry:
+            if isinstance(xml_entry[tagname], dict) and 'value' in xml_entry[tagname]:
+                value = xml_entry[tagname]['value']
+            elif isinstance(xml_entry[tagname], str):
+                value = xml_entry[tagname]
+
+        if value is None:
+            log.trace('Could not get value for: %s, type: %s', tagname, in_type)
+            return
+
+        if out_prefix_name is not None and not out_prefix_name.isspace():
+            tagname = out_prefix_name + name
+
+        if isinstance(value, in_type):
+            entry[tagname] = value
+        elif in_type == int or in_type == float:
+            number = str_to_number(value)
+            if number is not None:
+                entry[tagname] = number
+        elif in_type == datetime:
+            date = str_to_naive_utc(value)
+            if date is not None:
+                entry[tagname] = date
+        elif in_type == bool:
+            boolvalue = None
+            if isinstance(value, str):
+                if str_to_boolean(value):
+                    boolvalue = True
+            if not boolvalue:
+                number = str_to_number(value)
+                if number is not None and number > 0:
+                    boolvalue = True
+                else:
+                    boolvalue = False
+            entry[tagname] = boolvalue
+
+    def _fill_namespace_attributes(self, xml_entry, entry):
+        for key in NAMESPACE_ATTRIBUTE_MAP:
+            self._set_ns_attribute(key, xml_entry, entry, NAMESPACE_NAME, NAMESPACE_PREFIX, NAMESPACE_ATTRIBUTE_MAP[key])
```

Comment:
Mutable methods are harder to read, it would be probably much more readable if this could be done like

`entry[key] = self._parse_xml_entry(xml_entry, ....)`

Throw exception if value can not be parsed/set